src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1600,7 +1600,17 @@ private void AssignPatternVariables(BoundPattern pattern, bool definitely = true
                         break;
                     }
                 case BoundKind.DiscardPattern:
+                case BoundKind.TypePattern:
                     break;
+                case BoundKind.SlicePattern:
+                    {
+                        var pat = (BoundSlicePattern)pattern;
+                        if (pat.Pattern != null)
+                        {
+                            AssignPatternVariables(pat.Pattern);
```

Comment:
>AssignPatternVariables [](http://example.com/codeflow?start=28&length=22)

Does this need to pass `definitely`? Seems like `o is not { ..var x }` should be definitely-assigned when false.

Reply:
This got me curious. I had a hard time to find a case that gives unexpected result given we always pass def=true.

Turns out no tests fails with def=true for any patterns.

I think there's couple of reasons for this: (1) We never accept variables under `not` anywhere except if it's directly under `is`.
(2) We assign variables in the true state and eventually flip it for `is not`. 

The key point here is that we either have *all* variables assigned in a branch or none. So variables that *could* be assigned are assigned in either true or false state, and others are reported as an illegal designation. Note that nested `not` patterns are being handled in the `is` analysis, therefore we don't need to do anything extra in this method.

Reply:
I'd stick with it though. We need it to report usages of these already illegal variables in a switch case, for example.